src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -109,7 +109,9 @@ private IMethodNode CreateMethodEntrypointNode(MethodDesc method, TypeDesc const
             }
 
             MethodWithGCInfo localMethod = null;
-            if (CompilationModuleGroup.ContainsMethodBody(method, false))
+            // TODO-PERF: for now, we never emit native code for array methods as Crossgen ignores
+            // them too. At some point we might be able to "exceed Crossgen CQ" by adding this support.
+            if (CompilationModuleGroup.ContainsMethodBody(method, false) && !(method is ArrayMethod))
```

Comment:
Why not just make `ContainsMethodBody` not return a lie here? If we're not generating a method body for array methods, then the compilation group doesn't contain a body for them.

Reply:
I can easily move the check into ContainsMethodBody, that sounds reasonable to me.